Observer/ProductSaveAfter.php

```diff
@@ -0,0 +1,139 @@
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Helper\Webhook;
+use Klaviyo\Reclaim\Helper\Logger;
+use Klaviyo\Reclaim\Model\ProductsFactory;
+
+use Magento\Catalog\Model\CategoryFactory;
+use Magento\Catalog\Model\Product;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+
+class ProductSaveAfter implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * Klaviyo logger helper
+     * @var \Klaviyo\Reclaim\Helper\Logger $klaviyoLogger
+     */
+    protected $_klaviyoLogger;
+
+    /**
+     * @var Webhook $webhookHelper
+     */
+    protected $_webhookHelper;
+    protected $_categoryFactory;
+    protected $_klProductFactory;
+    protected $_stockRegistry;
+    protected $product_category_names = [];
+
+    /**
+     * @param Webhook $webhookHelper
+     * @param ScopeSetting $klaviyoScopeSetting
+     * @param CategoryFactory $categoryFactory
+     * @param ProductsFactory $klProductFactory
+     * @param StockRegistryInterface $stockRegistry
+     * @param Logger $klaviyoLogger
+     */
+    public function __construct(
+        Webhook $webhookHelper,
+        ScopeSetting $klaviyoScopeSetting,
+        CategoryFactory $categoryFactory,
+        ProductsFactory $klProductFactory,
+        StockRegistryInterface $stockRegistry,
+        Logger $klaviyoLogger
+    ) {
+        $this->_webhookHelper = $webhookHelper;
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+        $this->_categoryFactory = $categoryFactory;
+        $this->_klProductFactory = $klProductFactory;
+        $this->_klaviyoLogger = $klaviyoLogger;
+        $this->_stockRegistry = $stockRegistry;
+    }
+
+    /**
+     * customer register event handler
+     *
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(Observer $observer)
+    {
+        $product = $observer->getEvent()->getProduct();
+        $storeIds = $product->getStoreIds();
+        $storeIdKlaviyoMap = $this->_klaviyoScopeSetting->getStoreIdKlaviyoAccountSetMap($storeIds);
+
+        foreach ($storeIdKlaviyoMap as $klaviyoId => $storeIds) {
+            if (empty($storeIds)) {continue;}
+
+            if ($this->_klaviyoScopeSetting->getWebhookSecret() && $this->_klaviyoScopeSetting->getProductSaveAfterSetting($storeIds[0])) {
+              $normalizedProduct = $this->normalizeProduct($product);
+              $data = [
+                "status"=>"NEW",
+                "topic"=>"product/save",
+                "klaviyo_id"=>$klaviyoId,
+                "payload"=>json_encode($normalizedProduct)
+              ];
+              $klProduct = $this->_klProductFactory->create();
+              $klProduct->setData($data);
+              $klProduct->save();
```

Comment:
I don't think we need to create a new Model to save a new row
```suggestion
              $klProduct = $this->_klProductFactory->setData($data);
              $klProduct->save();
```

Reply:
i tried this suggested change but i got an error when it ran.

Reply:
scratch that, you're correct, my bad on this one!